Projects/Server/Main.cs

```diff
@@ -345,6 +396,12 @@ private static void HandleClosed()
         {
             EventSink.InvokeShutdown();
         }
+
+        if (_cyclesPerSecond != null)
+        {
+            CycleArrayPool.Return(_cyclesPerSecond);
```

Comment:
Are we trying to treat this like a circular buffer? If so, we should just treat it like that. Hold a static array (no need for an array pool), with a start and count pointer.